pulsar-client/src/main/java/com/yahoo/pulsar/client/api/PulsarClientException.java

```diff
@@ -175,4 +176,34 @@ public ChecksumException(String msg) {
             super(msg);
         }
     }
+    
+    public static class TopicNotFoundException extends PulsarClientException {
+        public TopicNotFoundException(String msg) {
+            super(msg);
+        }
+    }
+    
+    public static class ConsumerIdNotFoundException extends PulsarClientException {
+        public ConsumerIdNotFoundException(String msg) {
+            super(msg);
+        }
+    }
+    
+    public static class SubscriptionNotFoundException extends PulsarClientException {
```

Comment:
As per convention:-
For every error in ServerError is mapped to a unique exception in Java and a unique Result in C++ via clientCnx.getPulsarClientException() (Java) and clientConnection.getResult(serverError)

Since UnknownError is a ServerError type I mapped it to a separate ClientException